subprojects/language-java/src/test/groovy/org/gradle/api/tasks/compile/CompileOptionsTest.groovy

```diff
@@ -16,135 +16,205 @@
+    static final TEST_FORK_OPTION_MAP = [someForkOption: 'someForkOptionValue']
 
     CompileOptions compileOptions
 
-    @Before public void setUp()  {
+    def setup()  {
         compileOptions = new CompileOptions()
         compileOptions.debugOptions = [optionMap: {TEST_DEBUG_OPTION_MAP}] as DebugOptions
         compileOptions.forkOptions = [optionMap: {TEST_FORK_OPTION_MAP}] as ForkOptions
     }
 
-    @Test public void testCompileOptions() {
-        assertTrue(compileOptions.debug)
-        assertTrue(compileOptions.failOnError)
-        assertTrue(compileOptions.warnings)
-
-        assertFalse(compileOptions.deprecation)
-        assertFalse(compileOptions.listFiles)
-        assertFalse(compileOptions.verbose)
-        assertFalse(compileOptions.fork)
-
-        assertThat(compileOptions.compilerArgs, isEmpty())
-        assertNull(compileOptions.encoding)
-        assertNull(compileOptions.bootClasspath)
-        assertNull(compileOptions.extensionDirs)
-
-        assertNotNull(compileOptions.forkOptions)
-        assertNotNull(compileOptions.debugOptions)
+    @SuppressWarnings("GrDeprecatedAPIUsage")
+    def "default compile options"() {
+        expect:
+        compileOptions.debug
+        compileOptions.failOnError
+        compileOptions.warnings
+
+        !compileOptions.deprecation
+        !compileOptions.listFiles
+        !compileOptions.verbose
+        !compileOptions.fork
+
+        compileOptions.compilerArgs.empty
+        compileOptions.encoding == null
+        compileOptions.bootClasspath == null
+        compileOptions.bootstrapClasspath == null
+        compileOptions.extensionDirs == null
+
+        compileOptions.forkOptions != null
+        compileOptions.debugOptions != null
     }
 
-    @Test public void testOptionMapForDebugOptions() {
+    def "option map for debug options"() {
         Map optionMap = compileOptions.optionMap()
-        assertEquals(optionMap.subMap(TEST_DEBUG_OPTION_MAP.keySet()), TEST_DEBUG_OPTION_MAP)
-        assertEquals(optionMap.subMap(TEST_FORK_OPTION_MAP.keySet()), TEST_FORK_OPTION_MAP)
+        expect:
+        optionMap.subMap(TEST_DEBUG_OPTION_MAP.keySet()) == TEST_DEBUG_OPTION_MAP
+        optionMap.subMap(TEST_FORK_OPTION_MAP.keySet()) == TEST_FORK_OPTION_MAP
     }
 
-    @Test public void testOptionMapWithNullables() {
-        Map optionMap = compileOptions.optionMap()
-        Map nullables = [
-                encoding: 'encoding',
-                bootClasspath: 'bootClasspath',
-                extensionDirs: 'extdirs'
-        ]
-        nullables.each {String field, String antProperty ->
-            assertFalse(optionMap.keySet().contains(antProperty))
-        }
-
-        nullables.keySet().each {compileOptions."$it" = "${it}Value"}
-        optionMap = compileOptions.optionMap()
-        nullables.each {String field, String antProperty ->
-            assertEquals("${field}Value" as String, optionMap[antProperty])
-        }
+    @Unroll
```

Comment:
I prefer placing the `@Unroll` annotation on the class itself instead of on individual methods. WDYT?